### ISSUE TITLE
Issue with spectral loss standard dev / variance

### ISSUE THREAD

**Issue by vmos1** [vmos1/cosmogan_pytorch#1]
https://github.com/vmos1/cosmogan_pytorch/blob/b01806445aed11d90a5ba5692786009934bc8214/cosmogan/1_main_code/spec_loss.py#L282

The code currently measure standard error. The loss function will work only if the reference sample and generated sample have the same number of images. 
The better option is to use standard deviation. 

However, even that can cause issues due to the backward propagation of the square root function. 
And even better option is to use variance.

**Comment by vmos1** [comment]
the Nan issue can be caused by large learning rate (0.002).
Variance fix is in place
